WordPress/Classes/ViewRelated/System/Search/WPSearchControllerConfigurator.m

```diff
@@ -0,0 +1,112 @@
+#import "WPSearchControllerConfigurator.h"
+#import "WordPress-Swift.h"
+
+const CGFloat SearchBarWidth = 280.0;
+const CGFloat SearchBariPadWidth = 600.0;
+const CGFloat SearchWrapperViewPortraitHeight = 64.0;
+const CGFloat SearchWrapperViewLandscapeHeight = 44.0;
+const NSTimeInterval SearchBarAnimationDuration = 0.2; // seconds
+
+@interface WPSearchControllerConfigurator ()
+
+@property (nonatomic, weak) WPSearchController *searchController;
+@property (nonatomic, weak) UIView *searchWrapperView;
+@property (nonatomic, weak) UISearchBar *searchBar;
+
+@end
+
+@implementation WPSearchControllerConfigurator
+
+- (instancetype)initWithSearchController:(WPSearchController *)searchController
+                   withSearchWrapperView:(UIView *)searchWrapperView
+{
+    self = [super init];
+    if (self) {
+        _searchController = searchController;
+        _searchWrapperView = searchWrapperView;
+        _searchBar = _searchController.searchBar;
+    }
+    
+    return self;
+}
+
+- (void)configureSearchControllerBarAndWrapperViewOfClass:(Class)class
+{
+    [self configureSearchController];
+    [self configureSearchBar:class];
+    [self configureSearchWrapper];
+}
+
+- (void)configureSearchController
+{
+    self.searchController.dimsBackgroundDuringPresentation = NO;
+    self.searchController.hidesNavigationBarDuringPresentation = YES;
+}
+
+- (void)configureSearchBar:(Class)class
+{
+    [self configureSearchBarPlaceholder:class];
+    
+    self.searchBar.translatesAutoresizingMaskIntoConstraints = NO;
+    self.searchBar.accessibilityIdentifier = @"Search";
+    self.searchBar.autocapitalizationType = UITextAutocapitalizationTypeNone;
+    self.searchBar.backgroundImage = [[UIImage alloc] init];
+    self.searchBar.tintColor = [WPStyleGuide grey]; // cursor color
+    self.searchBar.translucent = NO;
+    self.searchBar.barStyle = UIBarStyleBlack;
+    self.searchBar.barTintColor = [WPStyleGuide wordPressBlue];
+    self.searchBar.showsCancelButton = YES;
+    [self.searchBar setImage:[UIImage imageNamed:@"icon-clear-textfield"] forSearchBarIcon:UISearchBarIconClear state:UIControlStateNormal];
+    [self.searchBar setImage:[UIImage imageNamed:@"icon-post-list-search"] forSearchBarIcon:UISearchBarIconSearch state:UIControlStateNormal];
+    
+    [self configureSearchBarForSearchView:class];
+}
+
+- (void)configureSearchBarPlaceholder:(Class)class
+{
+    // Adjust color depending on where the search bar is being presented.
+    UIColor *placeholderColor = [WPStyleGuide wordPressBlue];
+    NSString *placeholderText = NSLocalizedString(@"Search", @"Placeholder text for the search bar on the post screen.");
+    NSAttributedString *attrPlacholderText = [[NSAttributedString alloc] initWithString:placeholderText attributes:[WPStyleGuide defaultSearchBarTextAttributes:placeholderColor]];
+    [[UITextField appearanceWhenContainedIn:[UISearchBar class], class, nil] setAttributedPlaceholder:attrPlacholderText];
```

Comment:
It looks like the class is passed around just to set the appearance, yah?  I'm a bit torn and wonder if it would be okay to accept a little bit of duplication between the VC in this case. I've no real reason other than it just felt a bit weird to me. :shrug:  @diegoreymendez thoughts?